### PR TITLE
frontend fixes

### LINKED ISSUE #50: [FE] Closing the success modal by clicking outside of it doesnt dismiss it
This might also happen with the error modal, not sure.

But basically when you open the modal back up it still has that message in it

### PR DESCRIPTION
**Pull request by Zyrenth**
this fixes #50 and #49

### CODE REVIEW

**Comment by Govorunb** [comment]
> actually disable the redeem button when form is invalid

This removes the native feedback from the browser that was there before
image

I'd much rather see a fix for the endless pushing to the array instead

(sorry for the nit all the other stuff looks great)

**Comment by Alexejhero** [comment]
I agree with govo

File: frontend/www/src/modules/modal.ts [hunk 1]
```diff
@@ -191,6 +192,7 @@ function hideSuccessModal(closeMainModal = false) {
 
 function checkForm() {
     $modalConfirm.ariaDisabled = $modalOptionsForm.checkValidity() ? null : "";
+    $modalConfirm.disabled = !$modalOptionsForm.checkValidity();
```
Comment (Alexejhero) [inline frontend/www/src/modules/modal.ts:195]:
```suggestion
```

**Comment by Alexejhero** [comment]
Added in 972cf84

**Pull request closed without merging**